ui/app/AppLayouts/Profile/views/MyProfileView.qml

```diff
@@ -66,6 +66,24 @@ SettingsContentBase {
             Layout.fillWidth: true
             currentIndex: editPreviwTabBar.currentIndex
 
+            onImplicitHeightChanged: {
+                // During opening Profile, StackLayout always set implicitHeight of his biggest
+                // child implicitHeight and it reflecting on a scroll area.
+                // Use implicitHeight of current child
+                if (currentIndex === 0 && implicitHeight !== settingsView.implicitHeight) {
+                    implicitHeight = settingsView.implicitHeight
+                } else if (currentIndex === 1 && implicitHeight !== profilePreview.implicitHeight) {
+                    implicitHeight = profilePreview.implicitHeight
+                }
+            }
+
+            onCurrentIndexChanged: {
+                switch(currentIndex) {
+                case 0: implicitHeight = settingsView.implicitHeight; break
+                case 1: implicitHeight = profilePreview.implicitHeight; break
```

Comment:
Implicit values should never be assigned to

Reply:
Do I understand correctly that you suggest using `height` instead?

Reply:
Can I disagree with you?
Implicit size is supposed to be used when calculating the size of an item based on its contents. Height on a parent item may affect the size of its children and it should never be the case when you set implicit size.

Reply:
Agree with both. As I see it, the `implicit*` sizes are hints for clients on the content size of a resizable control and should be set from the component's implementation. Implicit sizes are taken into consideration as hints for layouting if `Layout` attached property hints are missing. In this case `Layout.prefferedWidth` and `Layout.prefferedHeight` have the same effect and show better the intention.

Reply:
```diff
diff --git a/ui/app/AppLayouts/Profile/views/MyProfileView.qml b/ui/app/AppLayouts/Profile/views/MyProfileView.qml
index e99f6581e..ef2f33575 100644
--- a/ui/app/AppLayouts/Profile/views/MyProfileView.qml
+++ b/ui/app/AppLayouts/Profile/views/MyProfileView.qml
@@ -63,24 +63,28 @@ SettingsContentBase {
         width: root.contentWidth
 
         StackLayout {
+            id: stackLayout
             Layout.fillWidth: true
             currentIndex: editPreviwTabBar.currentIndex
 
             MyProfileSettingsView {
                 id: settingsView
-                Layout.fillWidth: true
                 profileStore: root.profileStore
                 privacyStore: root.privacyStore
                 walletStore: root.walletStore
+
+                onVisibleChanged: if (visible) stackLayout.Layout.preferredHeight = Qt.binding(() => settingsView.implicitHeight)
+                Component.onCompleted: stackLayout.Layout.preferredHeight = Qt.binding(() => settingsView.implicitHeight)
             }
 
             MyProfilePreview {
                 id: profilePreview
-                Layout.fillWidth: true
                 profileStore: root.profileStore
                 contactsStore: root.contactsStore
                 dirtyValues: settingsView.dirtyValues
                 dirty: settingsView.dirty
+
+                onVisibleChanged: if (visible) stackLayout.Layout.preferredHeight = Qt.binding(() => profilePreview.implicitHeight)
             }
         }
```

@mprakhov what do you think about doing it in a more declarative way?

Reply:
this one looks really great. Refactored 👍